.config/ags/modules/dock/dock.js

```diff
@@ -177,82 +236,87 @@ const PinnedApps = () => Widget.Box({
+                // const hyprlandClients = JSON.parse(exec('hyprctl clients -j'));
+                // for (const index in hyprlandClients) {
+                //     const client = hyprlandClients[index];
+                //     const clientLeft = client.at[0];
+                //     const clientRight = client.at[0] + client.size[0];
+                //     const clientTop = client.at[1];
+                //     const clientBottom = client.at[1] + client.size[1];
+                //
+                //     if (client.workspace.id == currentWorkspace) {
+                //         if (
+                //             // clientLeft < dockRight &&
+                //             // clientRight > dockLeft &&
+                //             // clientTop < dockBottom &&
+                //             // clientBottom > dockTop
+                //         ) {
+                //             self.revealChild = false;
+                //             return;
+                //         }
+                //     }
+                // }
+                // // if (currentWorkspace === client.workspace.id) {
+                // self.revealChild = true;
+                // // }
+                self.revealChild = activeMonitorId() === monitor
             }
         },
         revealChild: false,
         transition: 'slide_up',
         transitionDuration: userOptions.animations.durationLarge,
         child: dockContent,
-        // setup: (self) => self
-        //     .hook(Hyprland, (self) => self.attribute.updateShow(self))
-        //     .hook(Hyprland.active.workspace, (self) => self.attribute.updateShow(self))
-        //     .hook(Hyprland.active.client, (self) => self.attribute.updateShow(self))
-        //     .hook(Hyprland, (self) => self.attribute.updateShow(self), 'client-added')
-        //     .hook(Hyprland, (self) => self.attribute.updateShow(self), 'client-removed')
-        // ,
+        setup: (self) => self
+            // .hook(Hyprland, (self) => self.attribute.updateShow(self))
+            .hook(Hyprland.active.workspace, (self) => self.attribute.updateShow(self))
+            // .hook(Hyprland.active.client, (self) => self.attribute.updateShow(self))
+            .hook(Hyprland, (self) => self.attribute.updateShow(self), 'client-added')
+            .hook(Hyprland, (self) => self.attribute.updateShow(self), 'client-removed')
+        ,
     })
     return EventBox({
         onHover: () => {
             dockRevealer.revealChild = true;
         },
-        onHoverLost: () => {
-            if (Hyprland.active.client.attribute.class.length === 0) return;
-            dockRevealer.revealChild = false;
-        },
+        // onHoverLost: () => {
+        //     if (Hyprland.active.client.attribute.class.length === 0) { return }
+        //     dockRevealer.revealChild = false;
+        // },
         child: Box({
             homogeneous: true,
-            css: 'min-height: 2px;',
+            css: 'min-height: 20px;',
```

Comment:
20px would be too much and might block apps. can you add a config option and have it default to 5? (since that's the hyprland outer gap value)